spec/index.md

```diff
@@ -2376,6 +2376,23 @@ This function returns fee percentiles, measured in millisatoshi/vbyte (1000 mill
 
 The [standard nearest-rank estimation method](https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method), inclusive, with the addition of a 0th percentile is used. Concretely, for any i from 1 to 100, the ith percentile is the fee with rank `⌈i * 100⌉`. The 0th percentile is defined as the smallest fee (excluding coinbase transactions).
 
+### IC method `bitcoin_get_block_headers` {#ic-bitcoin_get_block_headers}
+
+:::note
+
+The `bitcoin_get_block_headers` endpoint is considered EXPERIMENTAL. Canister developers must be aware that this endpoint may evolve in a non-backward-compatible way.
+
+:::
+
+Given a start height and an optional end height, the function returns the block headers in the provided range. The range is inclusive, i.e., the block headers at the start and end heights are returned as well.
+
+ If no end height is specified, all blocks until the tip height, i.e., the largest available height, are returned. However, if the range from the start height to the end height or the tip height is large, only a prefix of the requested block headers may be returned in order to bound the size of the response.
```

Comment:
What should happen if the end height is specified and it is larger than the height of the main chain?

Reply:
Good question. In my opinion, the most natural approach is to return block headers until the tip height in this case.
Do you agree?

I think the text should be updated to make this clear.

Reply:
In my opinion, it is clearer to return an error, if the user wants all blocks until the end it has an option to specify the end of the range as `None`. Furthermore, since we can limit the number of returned `block_headers` if the response is too large, it may happen following case, user request headers `[request_start, request_end]`, and the receivers `[request_start, response_end]` where `response_end < request_end` at that moment user cannot know if we have limited the size of response or bitcoin canister does not have rest of the block headers. So to distinguish two cases I will prefer to return an error.

Reply:
Yeah, that makes sense.
@ielashi @dsarlis, do you agree with @dragoljub-duric's proposal?

Reply:
The proposal SGTM